src/sentry/api/event_search.py

```diff
@@ -116,12 +118,28 @@ def visit_date_format(self, node, children):
 
     def visit_basic_filter(self, node, children):
         search_key, _, search_value = children
+
+        # if this matched with the `has` key, then their value for
+        # this was not a valid key pattern
+        if search_key == 'has':
```

Comment:
Why would this path be hit at all? Wouldn't `has_filter` happen first to prevent this?

Reply:
👍 done!